crates/generator/src/generator.rs

```diff
@@ -179,16 +253,36 @@ impl Generator {
             let mut machine = TraceMachine::new(default_machine);
 
             machine.load_program(&backend.generator, &[])?;
-            match machine.run() {
+            let maybe_ok = machine.run();
+            used_cycles = machine.machine.cycles();
+            drop(machine);
+
+            // Subtract tx cycles.
+            // NOTE: already subtract syscall cycles from cycles pool during machine running to
+            // interrupt execution eariler.
+            let used_syscall_cycles = cycles_pool_bak.cycles() - cycles_pool.cycles();
+            cycles_pool.sub_cycles(used_cycles.saturating_sub(used_syscall_cycles));
```

Comment:
Why minus the syscall cycles? By my understand the mem-pool should reject a tx if it consumes too many syscall cycles, it that right?

Reply:
Machine cycles include syscall cycles. And we already minus syscall cycles in syscalls implementation [here](
https://github.com/nervosnetwork/godwoken/pull/767/files#diff-16f67f4822618c8e4616f5d07b43084312bdba514bf01f1c696cd16a86a6d496R152)